python/cudf/cudf/core/buffer.py

```diff
@@ -5,108 +5,206 @@
 import functools
 import operator
 import pickle
-from typing import Any, Dict, Tuple
+from typing import (
+    Any,
+    Dict,
+    Mapping,
+    Protocol,
+    Tuple,
+    Union,
+    runtime_checkable,
+)
 
 import numpy as np
 
 import rmm
 
 import cudf
 from cudf.core.abc import Serializable
+from cudf.utils.string import format_bytes
+
+
+@runtime_checkable
+class DeviceBufferLike(Protocol):
+    def __getitem__(self, key: Union[int, slice]) -> DeviceBufferLike:
+        """Create a new view of the buffer."""
```

Comment:
Should we include ellipses on these methods/properties for style consistency with the [Python docs](https://docs.python.org/3/library/typing.html?#typing.Protocol)? (and consistency with other stub-like interfaces I've seen in `.pyi` files)
```suggestion
    def __getitem__(self, key: Union[int, slice]) -> DeviceBufferLike:
        """Create a new view of the buffer."""
        ...
```

Reply:
I don't think we should use `...` if we don't have to. Most stubs uses `...` because they don't have a docstring or implementation.

Reply:
Okay, sounds good.